ros1_bridge/__init__.py

```diff
@@ -596,27 +596,37 @@ def determine_common_services(
                 break
             for i, ros1_field in enumerate(ros1_fields[direction]):
                 ros1_type = ros1_field[0]
-                ros2_type = str(ros2_fields[direction][i].type)
+                ros2_type = ros2_fields[direction][i].type
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
+                if ros1_type != str(ros2_type) or ros1_name != ros2_name:
                     # if the message types have a custom mapping their names
                     # might not be equal, therefore check the message pairs
-                    if (ros1_type, ros2_type) not in message_string_pairs:
+                    if (ros1_type, str(ros2_type)) not in message_string_pairs:
                         match = False
                         break
+                if not ros2_type.array_size and not ros2_type.is_upper_bound:
+                    array_size = ros2_type.array_size if ros2_type.is_upper_bound else -1
+                else:
+                    array_size = None
+                if ros2_type.pkg_name:
+                    ros2_cpptype = ros2_type.pkg_name + "::msg::" + ros2_type.type
+                else:
+                    ros2_cpptype = str(ros2_type)
                 output[direction].append({
-                    'basic': False if '/' in ros1_type else True,
-                    'array': True if '[]' in ros1_type else False,
+                    'basic': not ros2_type.pkg_name,
+                    'array': ros2_type.is_array,
+                    'is_upper_bound': ros2_type.is_upper_bound,
+                    'array_size': array_size,
```

Comment:
Similar to my previous comment I suggest to avoid any kind of indirection and changing the semantic is existing names. Why not just pass `ros2_type.array_size` here?

Based on the above comment `array_size` is either `-1` or `None` here which doesn't seem to make much sense.